lib/modules/platform/gitlab/pr-cache.ts

```diff
@@ -0,0 +1,181 @@
+  ): Promise<GitlabPrCache> {
+    const res = new GitlabPrCache(repo, author, ignorePrAuthor);
+    const isSynced = memCache.get<true | undefined>('gitlab-pr-cache-synced');
+
+    if (!isSynced) {
+      await res.sync(http);
+      memCache.set('gitlab-pr-cache-synced', true);
+    }
+
+    return res;
+  }
+
+  private getPrs(): GitlabPr[] {
+    return this.items;
+  }
+
+  static async getPrs(
+    http: GitlabHttp,
+    repo: string,
+    author: string | null,
+    ignorePrAuthor: boolean,
+  ): Promise<GitlabPr[]> {
+    const prCache = await GitlabPrCache.init(
+      http,
+      repo,
+      author,
+      ignorePrAuthor,
+    );
+    return prCache.getPrs();
+  }
+
+  private setPr(pr: GitlabPr): void {
+    logger.debug(`Adding PR #${pr.number} to the PR cache`);
+    this.cache.items[pr.number] = pr;
+    this.updateItems();
+  }
+
+  static async setPr(
+    http: GitlabHttp,
+    repo: string,
+    author: string | null,
+    item: GitlabPr,
+    ignorePrAuthor: boolean,
+  ): Promise<void> {
+    const prCache = await GitlabPrCache.init(
+      http,
+      repo,
+      author,
+      ignorePrAuthor,
+    );
+    prCache.setPr(item);
+  }
+
+  private reconcile(rawItems: GitLabMergeRequest[]): boolean {
+    const { items: oldItems } = this.cache;
+    let { updated_at } = this.cache;
+
+    let needNextPage = true;
+
+    for (const rawItem of rawItems) {
+      const id = rawItem.iid;
+
+      const oldItem = oldItems[id];
+      const newItem = prInfo(rawItem);
+
+      const itemNewTime = DateTime.fromISO(rawItem.updated_at);
+
+      if (dequal(oldItem, newItem)) {
+        needNextPage = false;
+        continue;
+      }
+
+      oldItems[id] = newItem;
+
+      const cacheOldTime = updated_at ? DateTime.fromISO(updated_at) : null;
+      if (!cacheOldTime || itemNewTime > cacheOldTime) {
+        updated_at = rawItem.updated_at;
+      }
+    }
+
+    this.cache.updated_at = updated_at;
+    return needNextPage;
+  }
+
+  private async sync(http: GitlabHttp): Promise<GitlabPrCache> {
+    logger.debug('Syncing PR list');
+    const searchParams = {
+      per_page: this.items.length ? '20' : '100',
+    } as any;
+    if (!this.ignorePrAuthor) {
+      searchParams.scope = 'created_by_me';
+    }
+    let query: string | null = getQueryString(searchParams);
+
+    while (query) {
+      const res = await http.getJsonUnchecked<GitLabMergeRequest[]>(
+        `/projects/${this.repo}/merge_requests?${query}`,
+        {
+          memCache: false,
+        },
+      );
+
+      const needNextPage = this.reconcile(res.body);
+      if (!needNextPage) {
+        break;
+      }
+
+      const uri = parseUrl(parseLinkHeader(res.headers.link)?.next?.url);
+      if (uri && http.baseUrl) {
+        if (process.env.GITLAB_IGNORE_REPO_URL) {
+          const defaultEndpoint = new URL(http.baseUrl);
+          uri.protocol = defaultEndpoint.protocol;
+          uri.host = defaultEndpoint.host;
+        }
+      }
```

Comment:
wait, you're only using `search`, so this doesn't change anything on `query`
```suggestion
```

Reply:
😅